### PR TITLE
Support for bootstrap commands to use custom data for templates

### LINKED ISSUE #1100: Support for passing in custom template data when using a template file for the bootstrap and generate commands
Hi, we have a use case where currently we used Ginkgo for System Levels E2E API Tests. We leverage the `generate` and `bootstrap` commands with the `--template` to be able to get team members up and running coding new test specs. 

That said there are times where we want to modify our template with more variables to inject other metadata during rendering rather than hardcoding it in or modifying it afterwards. Below is an example where we would want to render in the values for `.TestSpecName`, `.FrameworkImport`, `.TestFeature`, `.TestComponent`, `.TestId` as part of the `generate` command of our test spec template.

If you guys are OK with this I would be interested in contributing this to the commands

```
package {{.Package }}

/* This was generated from a template file. Please feel free to update as necessary */

import (
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

    "fmt"
	"strings"
	"time"
    "encoding/json"
    "context"


    //framework imports edit as required
    {{ .FrameworkImport }}

)

var _ = framework.{{ .TestSpecName }}SuiteDescribe("{{ .TestSpecName }} tests", Label("{{ .TestFeature }}"), func() {

    /* note:
    - You may need to implement specific implementation of the service/domain you are trying to test if it
    not already there in the pkg/ packages
    */

	defer GinkgoRecover()
    var err error
    var f *framework.Framework
    // use 'f' to access common controllers or the specific service controllers within the framework
	BeforeAll(func() {
		// Initialize the tests controllers
		f, err = framework.NewFramework()
		Expect(err).NotTo(HaveOccurred())
	})



    Describe("{{ .TestSpecName }} scenario to test", Label("{{ .TestComponent }}"), func() {
        // Declare variables here.

        BeforeEach(func() {

        // Initialize variables here.
        // Assert setup here.

        })

        It("{{ .TestId }} does some test action", func() { 

        // Implement test and assertions here
        
        })
    
    })

})
```

### PR DESCRIPTION
**Pull request by Dannyb48**
* Includes updates to docs
 * Includes associated integration tests
 * Resolves #1100

### CODE REVIEW

**Comment by Dannyb48** [comment]
@onsi when you get a chance can you review this PR?

**Comment by onsi** [comment]
hey @Dannyb48 this looks great - thanks for putting it together and apologies for the delay on my end.  I'm back in town now :)

I only have a couple of comments that I'll leave on the commit, but then should be good to pull in.

File: ginkgo/generators/bootstrap_command.go [hunk 1]
```diff
@@ -25,6 +26,9 @@ func BuildBootstrapCommand() command.Command {
 			{Name: "template", KeyPath: "CustomTemplate",
 				UsageArgument: "template-file",
 				Usage:         "If specified, generate will use the contents of the file passed as the bootstrap template"},
+			{Name: "templatedata", KeyPath: "CustomTemplateData",
```
Comment (onsi) [inline ginkgo/generators/bootstrap_command.go:29]: could we make this `template-data` to be more consistent with the rest of Ginkgo's multi-word flags?  Thanks!

File: ginkgo/generators/generate_command.go [hunk 2]
```diff
@@ -28,6 +29,9 @@ func BuildGenerateCommand() command.Command {
 			{Name: "template", KeyPath: "CustomTemplate",
 				UsageArgument: "template-file",
 				Usage:         "If specified, generate will use the contents of the file passed as the test file template"},
+			{Name: "templatedata", KeyPath: "CustomTemplateData",
```
Comment (onsi) [inline ginkgo/generators/generate_command.go:32]: could we make this `template-data` to be more consistent with the rest of Ginkgo's multi-word flags?  Thanks!

Comment (Dannyb48) [reply ginkgo/generators/generate_command.go:32]: ack, will do

File: ginkgo/generators/bootstrap_command.go (hunk 1 above)
Comment (Dannyb48) [reply ginkgo/generators/bootstrap_command.go:29]: ack, will do

File: docs/index.md [hunk 3]
```diff
@@ -5081,7 +5081,17 @@ will generate a file named `PACKAGE_suite_test.go` and
...
 
-will generate a file named `SUBJECT_test.go` (or `PACKAGE_test.go` if `<SUBJECT>` is not provided).  Both generators support custom templates using `--template`.  Take a look at the [Ginkgo's CLI code](https://github.com/onsi/ginkgo/tree/master/ginkgo/ginkgo/generators) to see what's available in the template.
+will generate a file named `SUBJECT_test.go` (or `PACKAGE_test.go` if `<SUBJECT>` is not provided).  Both generators support custom templates using `--template` 
+and the option to provide extra custom data to be rendered into the template, besides the default values, using `--templatedata`. The custom data should be a well structured JSON file. When loaded into the template the custom data will be available to access in the template from the global key `.CustomData`. For example
+with a JSON file
+```json
+{ "suitename": "E2E",
+  "labels": ["fast", "parallel", "component"]}
+```
+The custom data can be could accessed like so:
```
[the remark this answers is not in the document]
Comment (Dannyb48) [reply docs/index.md:5091]: ack, done

**Comment by onsi** [comment]
LGTM thanks!

I'll merge this in and probably cut a release next week after i finish up a few other things i'm working on.

**Pull request merged**